pkg/dialer/factory.go

```diff
@@ -54,15 +58,24 @@ func (f *Factory) ClusterDialer(clusterName string) (dialer.Dialer, error) {
 	}, nil
 }
 
-func IsCloudDriver(cluster *v3.Cluster) bool {
-	return !cluster.Spec.Internal &&
-		cluster.Status.Driver != "" &&
-		cluster.Status.Driver != v32.ClusterDriverImported &&
-		cluster.Status.Driver != v32.ClusterDriverRKE &&
-		cluster.Status.Driver != v32.ClusterDriverK3s &&
-		cluster.Status.Driver != v32.ClusterDriverK3os &&
-		cluster.Status.Driver != v32.ClusterDriverRke2 &&
-		cluster.Status.Driver != v32.ClusterDriverRancherD
+func IsPublicCloudDriver(cluster *v3.Cluster) bool {
+	if cluster.Spec.Internal ||
+		cluster.Status.Driver == "" ||
+		cluster.Status.Driver == v32.ClusterDriverImported ||
+		cluster.Status.Driver == v32.ClusterDriverRKE ||
+		cluster.Status.Driver == v32.ClusterDriverK3s ||
+		cluster.Status.Driver == v32.ClusterDriverK3os ||
+		cluster.Status.Driver == v32.ClusterDriverRke2 ||
+		cluster.Status.Driver == v32.ClusterDriverRancherD {
+		return false
+	}
```

Comment:
could this not be simplified to cluster.Status.Driver != EKS {return false}, or is it possible the driver is not set yet?

Reply:
The original was written this way. This is to protect against new drivers being added (like GKE or AKS). That is, if a new cloud driver is added, then we don't have to come here to alter this code as well.

Reply:
Maybe we should keep a slice of included or excluded drivers and just have a function that iterates through them, we could put it in a package for the cloud providers.